platform/http_client.hpp

```diff
@@ -129,7 +128,7 @@ class HttpClient
   // Cookies set by the client before request is run.
   std::string m_cookies;
   Headers m_headers;
-  bool m_handleRedirects = true;
+  bool m_followRedirects = false; // If true then in case of HTTP response 3XX make another request to follow redirected URL
```

Comment:
Won't changing the default value break some other existing requests?

Reply:
In `http_client_apple.mm` I replaced

```
if (_handleRedirects && response.statusCode >= 300 && response.statusCode < 400)
```

with

```
if (!_followRedirects && response.statusCode >= 300 && response.statusCode < 400)
```

Reply:
For Android it's tricky. In `android/app/src/main/cpp/app/organicmaps/util/HttpClient.cpp` there was a line:

```
SetBoolean(env, httpParamsObject.get(), ids.GetId("followRedirects"), m_handleRedirects);
```
Meaning that `m_handleRedirects` field is used as `followRedirects` field which is wrong. But HTTP Client in Java works in such way that it doesn't follow redirect to unknown protocols.
So if `followRedirects = true` and HTTP Response redirect to `om://oauth2/osm/callback?code=P7dyHlhtepaYT` then Java doesn't follow this redirect and everything works fine. So I changed code to:

```
SetBoolean(env, httpParamsObject.get(), ids.GetId("followRedirects"), m_followRedirects);
```

It should work correctly now.

Reply:
Tests are green, so this flag is handled correctly.